include/bluetooth/mesh/gen_loc.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_gen_loc Generic Location models
+ * @{
+ * @brief API for the Generic Location models.
+ */
+
+#ifndef BT_MESH_GEN_LOC_H__
+#define BT_MESH_GEN_LOC_H__
+
+#include <bluetooth/mesh.h>
+
+/**
+ * @defgroup gen_loc_altitude_defines Defines for altitude.
+ * Boundaries and special values for the altitude.
+ * @{
+ */
+/** Highest altitude that can be represented */
+#define GEN_LOC_ALTITUDE_MAX 32765
+/** Altitude is unknown or not configured. */
+#define GEN_LOC_ALTITUDE_UNKNOWN 0x7fff
+/** Altitude is out of bounds. */
+#define GEN_LOC_ALTITUDE_TOO_LARGE 0x7ffe
+/** @} */
+
+/**
+ * @defgroup gen_loc_floor_number_defines Defines for floor number.
+ * Boundaries and special values for the local floor number.
+ * @{
+ */
+/** Lowest floor number that can be represented. */
+#define GEN_LOC_FLOOR_NUMBER_MIN (-20)
+/** Highest floor number that can be represented. */
+#define GEN_LOC_FLOOR_NUMBER_MAX 232
+/** Ground floor (where the ground floor is referred to as floor 0) */
+#define GEN_LOC_FLOOR_NUMBER_GROUND_FLOOR_0 0xfc
+/** Ground floor (where the ground floor is referred to as floor 1) */
+#define GEN_LOC_FLOOR_NUMBER_GROUND_FLOOR_1 0xfd
+/** Unknown or unconfigured floor number. */
+#define GEN_LOC_FLOOR_NUMBER_UNKNOWN 0xff
+/** @} */
+
+/** Global location parameters. */
+struct gen_loc_global {
+	/** Global WGS84 North coordinate in degrees. */
+	double latitude;
+	/** Global WGS84 East coordinate in degrees. */
+	double longitude;
+	/**
+	 * Global altitude above the WGS84 datum in meters.
+	 * @sa gen_loc_altitude_defines
+	 */
+	s16_t altitude;
+};
+
+/** Local location parameters. */
+struct gen_loc_local {
+	/** Local north position in decimeters. */
+	s16_t north;
+	/** Local east position in decimeters. */
+	s16_t east;
+	/** Local altitude in decimeters. @sa gen_loc_altitude_defines */
+	s16_t altitude;
+	/** Floor number. @sa gen_loc_floor_number_defines */
+	s16_t floor_number;
+	/** Whether the device is movable. */
+	bool is_mobile;
+	/** Time since the previous position update, @ref K_FOREVER. */
```

Comment:
What is the time unit for time-related fields in these structures? Looks like it is in milliseconds.

Reply:
All timing is in milliseconds in zephyr/ncs, unless otherwise specified. The _ms suffix is never used, so I don't want to break convention, as that could cause confusion in all the cases that don't specify units.